src/tests/test_x509_tn_auth_list.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+* (C) 2006,2011,2012,2014,2015 Jack Lloyd
+* (C) 2022 René Meusel, Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "tests.h"
+
+#if defined(BOTAN_HAS_X509_CERTIFICATES)
+   #include <botan/data_src.h>
+   #include <botan/x509_ext.h>
+   #include <botan/x509cert.h>
+#endif
+
+namespace Botan_Tests {
+
+#if defined(BOTAN_HAS_X509_CERTIFICATES) && defined(BOTAN_TARGET_OS_HAS_FILESYSTEM)
+
+class X509test_TNAuthList_Tests final : public Test {
+   public:
+      std::vector<Test::Result> run() override {
+         /* cert with TNAuthList extension data generated by asn1parse cfg:
+
+            asn1=SEQUENCE:tn_auth_list
+
+            [tn_auth_list]
+            spc=EXP:0,IA5:1001
+            range=EXP:1,SEQUENCE:TelephoneNumberRange
+            one=EXP:2,IA5:333
+
+            [TelephoneNumberRange]
+            start1=IA5:111
+            count1=INT:128
+            start2=IA5:222
+            count2=INT:256
+          */
+         const std::string filename("TNAuthList.pem");
+
+         std::vector<Test::Result> results;
+         Test::Result result("X509test TNAuthList parsing");
+         result.start_timer();
+
+         Botan::X509_Certificate cert(Test::data_file("x509/x509test/" + filename));
+
+         using Botan::Cert_Extension::TNAuthList;
+
+         auto tn_auth_list = cert.v3_extensions().get_extension_object_as<TNAuthList>();
+
+         auto& tn_entries = tn_auth_list->get_entries();
+
+         result.confirm("cert has TNAuthList extension", tn_auth_list != nullptr, true);
+
+         result.test_eq(
+            "spc entry type", tn_entries[0].get_type() == TNAuthList::TNEntry::TN_ServiceProviderCode, true);
+         result.test_eq("spc entry data", tn_entries[0].getServiceProviderCode(), "1001");
+
+         result.test_eq(
+            "range entry type", tn_entries[1].get_type() == TNAuthList::TNEntry::TN_TelephoneNumberRange, true);
+         auto& range = tn_entries[1].getTelephoneNumberRange();
+         result.test_eq("range entries count", range.size(), 2);
+         result.test_eq("range entry 0 start data", range[0].start.value(), "111");
+         result.test_eq("range entry 0 count data", range[0].count, 128);
+         result.test_eq("range entry 1 start data", range[1].start.value(), "222");
+         result.test_eq("range entry 1 count data", range[1].count, 256);
+
+         result.test_eq("one entry type", tn_entries[2].get_type() == TNAuthList::TNEntry::TN_TelephoneNumber, true);
+         result.test_eq("one entry data", tn_entries[2].getTelephoneNumber(), "333");
+
+         results.push_back(result);
+         result.end_timer();
+         return results;
+      }
+};
+
+BOTAN_REGISTER_TEST("x509", "x509_tn_auth_list", X509test_TNAuthList_Tests);
```

Comment:
Thanks for the test case! I think this could live in `unit_x509.cpp`, though. For instance, look at the `test_x509_authority_information_access()` test for an example. Also removes the need of having a `std::vector<Test::Result>`. Simply return your single test result from a free-standing test function and call/push-back it in `X509_Cert_Unit_Tests::run()`.

Reply:
moved to the `unit_x509.cpp`